scripts/run_lighthouse_tests.py

```diff
@@ -50,11 +62,143 @@ def run_lighthouse_checks():
         python_utils.PRINT(line[:-1])
 
 
-def main():
+def start_google_app_engine_server(prodEnv):
+    """Start the Google App Engine server."""
+    if prodEnv:
+        app_yaml_filepath = 'app.yaml'
+    else:
+        app_yaml_filepath = 'app_dev.yaml'
+
+    python_utils.PRINT('Starting oppia server...')
+
+    p = subprocess.Popen(
+        '%s %s/dev_appserver.py --host 0.0.0.0 --port %s '
+        '--clear_datastore=yes --dev_appserver_log_level=critical '
+        '--log_level=critical --skip_sdk_update_check=true %s' % (
+            common.CURRENT_PYTHON_BIN, common.GOOGLE_APP_ENGINE_HOME,
+            APP_ENGINE_PORT, app_yaml_filepath), shell=True)
+    RUNNING_PROCESSES.append(p)
+
+
+def download_and_install_nginx():
+    """Download and install nginx."""
+    python_utils.PRINT('Installing nginx...')
+    update_command = ['sudo', 'apt-get', 'update']
+    install_command = ['sudo', 'apt-get', 'install', 'nginx']
+
+    subprocess.check_call(update_command)
+    subprocess.check_call(install_command)
+
+
+def wait_for_port_to_be_open(port_number):
+    """Wait until the port is open.
+
+    Args:
+        port_number: int. The port number to wait.
+    """
+    waited_seconds = 0
+    while (not common.is_port_open(port_number) and
+           waited_seconds < MAX_WAIT_TIME_FOR_PORT_TO_OPEN_SECS):
+        time.sleep(1)
+        waited_seconds += 1
+    if (waited_seconds ==
+            MAX_WAIT_TIME_FOR_PORT_TO_OPEN_SECS and
+            not common.is_port_open(port_number)):
+        python_utils.PRINT(
+            'Failed to start server on port %s, exiting ...' % port_number)
+        sys.exit(1)
+
+
+def start_proxy_server():
+    """Start the nginx proxy server."""
+    nginx_conf_file = os.path.join(
+        common.OPPIA_TOOLS_DIR, 'nginx.conf')
+    with python_utils.open_file(nginx_conf_file, 'w') as f:
+        f.write("""
+            events {
+
+            }
+            http {
+                server {
+                    listen 9999;
+
+                    gzip                on;
+                    gzip_buffers        16 8k;
+                    gzip_comp_level     9;
+                    gzip_http_version   1.1;
+                    gzip_proxied        any;
+                    gzip_types          *;
+                    gzip_vary           on;
+
+                    location / {
+                        proxy_set_header x-real-IP $remote_addr;
+                        proxy_set_header x-forwarded-for $proxy_add_x_forwarded_for;
+                        proxy_set_header host $host;
+                        proxy_pass http://127.0.0.1:8181;
+                    }
+                }
+            }
+        """)
+
+    python_utils.PRINT('Starting proxy server...')
+    start_server_command = ['sudo', 'nginx', '-c', nginx_conf_file]
```

Comment:
Yes
I noticed some issues with running nginx without sudo

Reply:
Can you please check how to run this without `sudo` and don't  have an error?

/cc @seanlip wanted to confirm that we should not use `sudo` iin any case for these scripts, right?

Reply:
@jameesjohn, any update on this?

Reply:
It would definitely be better to avoid doing so, if possible. I don't believe we use sudo anywhere in our scripts, and doing so tends to hide issues. @jameesjohn could you please be more specific about what issues you are running into?

Also, if sudo is really needed, then you probably need to let the user know in advance -- and have them confirm explicitly that they're fine with sudo before continuing to run the script.

Also I don't understand ... how are tests going to run on Travis if you use sudo?